diff -up elk-4.3.6/src/libxcifc.f90.libxc4 elk-4.3.6/src/libxcifc.f90
--- elk-4.3.6/src/libxcifc.f90.libxc4	2017-03-28 21:23:20.000000000 +0200
+++ elk-4.3.6/src/libxcifc.f90	2017-10-23 15:56:29.607689801 +0200
@@ -173,7 +173,7 @@ do k=2,3
 !------------------------------!
 ! set Tran-Blaha '09 constant if required
       if (id.eq.XC_MGGA_X_TB09) then
-        if (present(c_tb09)) call xc_f90_mgga_x_tb09_set_par(p,c_tb09)
+        if (present(c_tb09)) call xc_f90_func_set_ext_params(p,c_tb09)
       end if
       if (k.eq.2) then
 ! exchange
@@ -325,13 +325,13 @@ integer, intent(out) :: xcgrad
 logical, intent(inout) :: hybrid
 real(8), intent(inout) :: hybridc
 ! local variables
-integer j,k,id
+integer j,k,l,id
 integer fmly,flg
 character(256) name
 type(xc_f90_pointer_t) p,info
 ! check version is compatible (2.x)
-call xc_f90_version(j,k)
-if (j.ne.2) then
+call xc_f90_version(j,k,l)
+if (j.ne.3) then
   write(*,*)
   write(*,'("Error(xcdata_libxc): incompatible libxc version : ",I2.2,".",&
    &I3.3)') j,k
