--- elk-8.5.10/src/libxcifc.f90	2022-09-21 11:58:34.464765000 +0000
+++ elk-8.5.10/src/libxcifc.f90.libxc6	2022-11-07 22:30:17.202785243 +0000
@@ -418,7 +418,7 @@
 type(xc_f90_func_info_t) info
 ! check version is compatible
 call xc_f90_version(libxcv(1),libxcv(2),libxcv(3))
-if (libxcv(1).ne.5) then
+if (libxcv(1).gt.6) then
   write(*,*)
   write(*,'("Error(xcdata_libxc): incompatible Libxc version : ",I2.2,".",&
    &I3.3,".",I3.3)') libxcv(:)
