diff -up elk-3.3.17/src/libxcifc.f90.libxc elk-3.3.17/src/libxcifc.f90
--- elk-3.3.17/src/libxcifc.f90.libxc	2016-01-22 13:09:06.000000000 +0100
+++ elk-3.3.17/src/libxcifc.f90	2016-04-22 06:09:36.106899329 +0200
@@ -325,13 +325,13 @@ integer, intent(out) :: xcgrad
 logical, intent(inout) :: hybrid
 real(8), intent(inout) :: hybridc
 ! local variables
-integer j,k,id
+integer j,k,l,id
 integer fmly,flg
 character(256) name
 type(xc_f90_pointer_t) p,info
 ! check version is compatible (2.x)
-call xc_f90_version(j,k)
-if (j.ne.2) then
+call xc_f90_version(j,k,l)
+if (j.ne.3) then
   write(*,*)
   write(*,'("Error(xcdata_libxc): incompatible libxc version : ",I2.2,".",&
    &I3.3)') j,k
