--- src/libxcifc.f90.orig	2020-06-29 08:02:27.876149000 +0000
+++ src/libxcifc.f90	2020-09-07 12:58:56.658273737 +0000
@@ -19,6 +19,8 @@
  grho2,gup2,gdn2,gupdn,tau,tauup,taudn,ex,ec,vx,vc,vxup,vxdn,vcup,vcdn,dxdgr2, &
  dxdgu2,dxdgd2,dxdgud,dcdgr2,dcdgu2,dcdgd2,dcdgud,dxdg2r,dxdg2u,dxdg2d,dcdg2r, &
  dcdg2u,dcdg2d,wx,wxup,wxdn,wc,wcup,wcdn)
+! !USES:
+use, intrinsic :: iso_c_binding
 ! !INPUT/OUTPUT PARAMETERS:
 !   xctype : type of exchange-correlation functional (in,integer(3))
 !   n      : number of density points (in,integer)
@@ -98,7 +100,7 @@
 real(8), optional, intent(out) :: wc(n),wcup(n),wcdn(n)
 ! local variables
 integer nspin,fmly,id,k
-integer(8) np
+integer(c_size_t) np
 type(xc_f90_func_t) p
 real(8) ta
 ! allocatable arrays
@@ -312,6 +314,7 @@
 end subroutine
 
 subroutine fxcifc_libxc(fxctype,n,rho,rhoup,rhodn,fxc,fxcuu,fxcud,fxcdd)
+use, intrinsic :: iso_c_binding
 implicit none
 ! mandatory arguments
 integer, intent(in) :: fxctype(3),n
@@ -320,7 +323,7 @@
 real(8), optional, intent(out) :: fxc(n),fxcuu(n),fxcud(n),fxcdd(n)
 ! local variables
 integer nspin,fmly,id,k
-integer(8) np
+integer(c_size_t) np
 type(xc_f90_func_t) p
 ! allocatable arrays
 real(8), allocatable :: r(:,:),f(:,:)
